src/components/Popover/Popover.tsx

```diff
@@ -0,0 +1,247 @@
+import * as React from 'react';
+import styled from 'styled-components';
+import { Placement } from '@popperjs/core/lib/enums';
+import { usePopper } from 'react-popper';
+
+import { theme } from '../../essentials/theme';
+import { get } from '../../utils/themeGet';
+import { Colors, Spaces } from '../../essentials';
+import { ChevronDownIcon, ChevronUpIcon } from '../../icons/index';
+import { useClickOutside } from '../../utils/hooks/useClickOutside';
+
+import { Text } from '../Text/Text';
+
+import { PopoverContent } from './PopoverContent';
+
+interface PopoverRefObjectProps {
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    ref: any;
+}
+
+const PopoverTrigger = styled.div.attrs({ theme })<PopoverRefObjectProps>`
+    display: inline-block;
+    width: fit-content;
+    border-radius: ${get('radii.2')};
+`;
+
+const DefaultPopoverWrapper = styled.div.attrs({ theme })`
+    position: relative;
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    border: 1px solid ${get('semanticColors.button.secondary.borderHover')};
+    padding: 0.8125rem ${Spaces[2]};
+    border-radius: ${get('radii.2')};
+
+    &:hover {
+        cursor: pointer;
+        background-color: ${get('semanticColors.background.secondary')} !important;
+    }
+`;
+
+const PopoverContentContainer = styled.div<PopoverRefObjectProps>`
+    display: inline-block;
+`;
+
+const PopoverContentWrapper = styled.div`
+    display: inline-block;
+    box-sizing: border-box;
+    width: auto;
+    height: auto;
+    z-index: 1000;
+    box-shadow: ${get('shadows.small')};
+    max-height: none;
+
+    &:focus {
+        outline: 0;
+    }
+`;
+
+const KEY_CODE_MAP = {
+    ENTER: 13,
+    SPACE: 32,
+    ESC: 27
+};
+
+interface PopoverProps {
+    /**
+     * Popover Trigger (Only use Text, Link, Button or Icon component from @wave as a trigger)
+     */
+    children: React.ReactNode;
+    /**
+     * Popover content (can be any valid React Element or component containing React Elements)
+     */
+    content: React.ReactNode;
+    /**
+     * Optional: Specify the Popover content placement (placement changes automatically, if Popover content cannot fit inside viewport with selected placement)
+     */
+    placement?: Placement;
+    /**
+     * Optional: Specify the Popover content offset (margin between Popover trigger and content)
+     */
+    offset?: number;
+    /**
+     * Optional: Render popover content open by default
+     */
+    isOpen?: boolean;
+    /**
+     * Optional: Define a callback for when Popover content is opened
+     */
+    onOpen?: () => void;
+    /**
+     * Optional: Define a callback for when Popover content is closed
+     */
+    onClose?: () => void;
+}
+
+export const Popover: React.FC<PopoverProps> = ({
```

Comment:
Maybe this is a topic to discuss on a different issue, I think we should avoid using `React.FC` if we plan on upgrading React to 18.
Check out [this PR for the breaking changes in the types](https://github.com/DefinitelyTyped/DefinitelyTyped/pull/56210)

My opinion is that we should ideally only need to type the props, but at the moment we have a linter rule that does not allow that, so maybe for now we can type it as JSX.Element and then remove that rule in a different PR/issue?

Reply:
Yeah, I'm aware of this and of same opinion, but right now we still have a rule requiring it :) Also I agree JSX.Element would be better, however, at this point I think it's maybe better to be consistent and use React.FC same as all other @wave components? (considering we will be removing this from all components eventually)

Reply:
Sounds reasonable